### ISSUE TITLE
Files skipped while installing through PackageManager, due to errors in filename.

### ISSUE THREAD

**Issue by embicoin** [oferei/sublime-unity-completions#6]
In the Troubleshooting section you mention:
"Package Control may get stuck while installing the library. If that happens you'd see the animation running at the bottom indefinitely ..."

"This seems to be a bug in Package Control. I'm not sure what causes it - perhaps the relatively large size of this package."

I tried to investigate what was causing the error, hitting Sublime Text -> View -> Show Console, and then I had this output while the Package Control got stuck:

    Package Control: Skipping file from package named Unity Completions due to an invalid filename
    Package Control: Skipping file from package named Unity Completions due to an invalid filename
    Package Control: Skipping file from package named Unity Completions due to an invalid filename
    Package Control: Skipping file from package named Unity Completions due to an invalid filename
    reloading Packages/User/Package Control.sublime-settings

The last line was just the succesful exit, the package was finally installed (like 3 minutes after).

What files are causing this issue? It seems that it would be a lot faster if you fix those filenames.

**Comment by mattisx** [comment]
I'm getting the exact same issue as @embicoin (using Sublime Text 3). Cleared and restarted multiple times. Been working 10 minutes this time. Going to try the light version.

Thanks for your work with these plugins, much love!

**Comment by oferei** [comment]
Thanks! it's love that's makes this possible.  :)
enjoy the light package. I've just recently updated both packages to the new Unity. enjoy.

**Comment by Ivanca** [comment]
Thanks for the hard work @oferei .

I also want to mention that the package **does have invalid names**, at least for Windows because it doesn't allow question marks (?) on file names, so all the following files are being ignored:

```
CS/Other Classes/Array(?).sublime-snippet
Boo/Other Classes/Array(?).sublime-snippet
JavaScript/Other Classes/Array(?).sublime-snippet
CS/Other Classes/Array(?)_lower.sublime-snippet
Boo/Other Classes/Array(?)_lower.sublime-snippet
JavaScript/Other Classes/Array(?)_lower.sublime-snippet
CS/Other Classes/Path.Combine(?,_?)_func.sublime-snippet
Boo/Other Classes/Path.Combine(?,_?)_func.sublime-snippet
CS/Other Classes/Hashtable.Add(?,_?)_func.sublime-snippet
CS/Other Classes/Hashtable.Remove(?)_func.sublime-snippet
CS/Other Classes/Path.Combine(?,_?)_class.sublime-snippet
CS/Other Classes/Path.GetFileName(?)_func.sublime-snippet
Boo/Other Classes/Hashtable.Add(?,_?)_func.sublime-snippet
Boo/Other Classes/Hashtable.Remove(?)_func.sublime-snippet
Boo/Other Classes/Path.Combine(?,_?)_class.sublime-snippet
Boo/Other Classes/Path.GetFileName(?)_func.sublime-snippet
CS/Other Classes/Hashtable.Add(?,_?)_class.sublime-snippet
CS/Other Classes/Hashtable.Remove(?)_class.sublime-snippet
CS/Other Classes/Path.GetExtension(?)_func.sublime-snippet
CS/Other Classes/Path.GetFileName(?)_class.sublime-snippet
Boo/Other Classes/Hashtable.Add(?,_?)_class.sublime-snippet
Boo/Other Classes/Hashtable.Remove(?)_class.sublime-snippet
Boo/Other Classes/Path.GetExtension(?)_func.sublime-snippet
Boo/Other Classes/Path.GetFileName(?)_class.sublime-snippet
JavaScript/Other Classes/Path.Combine(?,_?)_func.sublime-snippet
CS/Other Classes/Hashtable.Contains(?)_func.sublime-snippet
CS/Other Classes/Path.GetExtension(?)_class.sublime-snippet
Boo/Other Classes/Hashtable.Contains(?)_func.sublime-snippet
Boo/Other Classes/Path.GetExtension(?)_class.sublime-snippet
JavaScript/Other Classes/Hashtable.Add(?,_?)_func.sublime-snippet
JavaScript/Other Classes/Hashtable.Remove(?)_func.sublime-snippet
JavaScript/Other Classes/Path.Combine(?,_?)_class.sublime-snippet
JavaScript/Other Classes/Path.GetFileName(?)_func.sublime-snippet
CS/Runtime Classes/Joint.OnJointBreak(?)_func.sublime-snippet
Boo/Runtime Classes/Joint.OnJointBreak(?)_func.sublime-snippet
CS/Other Classes/Hashtable.Contains(?)_class.sublime-snippet
Boo/Other Classes/Hashtable.Contains(?)_class.sublime-snippet
JavaScript/Other Classes/Hashtable.Add(?,_?)_class.sublime-snippet
JavaScript/Other Classes/Hashtable.Remove(?)_class.sublime-snippet
JavaScript/Other Classes/Path.GetExtension(?)_func.sublime-snippet
JavaScript/Other Classes/Path.GetFileName(?)_class.sublime-snippet
CS/Runtime Classes/Joint.OnJointBreak(?)_class.sublime-snippet
Boo/Runtime Classes/Joint.OnJointBreak(?)_class.sublime-snippet
JavaScript/Other Classes/Hashtable.Contains(?)_func.sublime-snippet
JavaScript/Other Classes/Path.GetExtension(?)_class.sublime-snippet
CS/Other Classes/Hashtable.ContainsKey(?)_func.sublime-snippet
CS/Other Classes/Path.Combine(?,_?)_func_lower.sublime-snippet
JavaScript/Runtime Classes/Joint.OnJointBreak(?)_func.sublime-snippet
Boo/Other Classes/Hashtable.ContainsKey(?)_func.sublime-snippet
Boo/Other Classes/Path.Combine(?,_?)_func_lower.sublime-snippet
```

**Comment by oferei** [comment]
Thanks Ivanca. I will make it right.

**Comment by oferei** [comment]
fixed in v5.0.2
